modules/core/src/main/java/org/apache/ignite/internal/cache/query/index/IndexProcessor.java

```diff
@@ -527,14 +529,35 @@ public void destroyOrphanIndex(
         tree.destroy();
     }
 
+    /**
+     * @return {@code true} In case of use an unwrapped PK for the index.
+     */
+    public boolean useUnwrappedPk(GridCacheContext<?, ?> cctx, String treeName) {
+        IgniteCacheDatabaseSharedManager db = cctx.shared().database();
+        db.checkpointReadLock();
+
+        try {
+            RootPage page = cctx.offheap().findRootPageForIndex(cctx.cacheId(), treeName, 0);
+
+            return page == null || metaPageInfo(page, cctx.groupId(), cctx.dataRegion().pageMemory()).useUnwrappedPk();
+        }
+        catch (IgniteCheckedException ignore) {
+            return true;
+        }
+        finally {
+            db.checkpointReadUnlock();
+        }
+
+    }
+
     /**
      * @param page Root page.
      * @param grpId Cache group id.
      * @param pageMemory Page memory.
-     * @return Inline size.
+     * @return Meta page info.
      * @throws IgniteCheckedException If something went wrong.
      */
-    private int inlineSize(RootPage page, int grpId, PageMemory pageMemory) throws IgniteCheckedException {
+    private MetaPageInfo metaPageInfo(RootPage page, int grpId, PageMemory pageMemory) throws IgniteCheckedException {
```

Comment:
Currently we have 2 similar methods for getting MetaPageInfo. One is here `IndexProcessor#metaPageInfo`, and another one is in `InlineIndexTree#metaInfo`. Let's use single one instead.